gcputil/credentials.go

```diff
@@ -141,38 +143,45 @@ func PublicKey(pemString string) (interface{}, error) {
 	return cert.PublicKey, nil
 }
 
-// OAuth2RSAPublicKey returns the PEM key file string for Google Oauth2 public cert for the given 'kid' id.
-func OAuth2RSAPublicKey(kid, oauth2BasePath string) (interface{}, error) {
-	oauth2Client, err := googleoauth2.New(cleanhttp.DefaultClient())
+func ServiceAccountPublicKey(serviceAccount string, keyId string) (interface{}, error) {
+	keyUrl := fmt.Sprintf(serviceAccountPublicKeyUrlTemplate, url.PathEscape(serviceAccount))
+	res, err := cleanhttp.DefaultClient().Get(keyUrl)
 	if err != nil {
-		return "", err
+		return nil, err
+	}
+	if err := googleapi.CheckResponse(res); err != nil {
+		return nil, err
 	}
 
-	if len(oauth2BasePath) > 0 {
-		oauth2Client.BasePath = oauth2BasePath
+	jwks := map[string]interface{}{}
+	if err := json.NewDecoder(res.Body).Decode(&jwks); err != nil {
+		return nil, fmt.Errorf("unable to decode JSON response: %v", err)
+	}
+	k, ok := jwks[keyId]
+	if !ok {
+		return nil, fmt.Errorf("service account %q key %q not found at GET %q", keyId, serviceAccount, keyUrl)
 	}
+	return PublicKey(k.(string))
```

Comment:
We may want to check the result of this type assertion  `kStr, ok := k.(string) // if !ok {...}`